controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/VersionStatusUpdater.java

```diff
@@ -27,10 +33,30 @@ protected void maintain() {
         try {
             VersionStatus newStatus = VersionStatus.compute(controller());
             controller().updateVersionStatus(newStatus);
+            newStatus.systemVersion().ifPresent(version -> {
+                try {
+                    controller().serviceRegistry().systemMonitor().reportSystemVersion(version.versionNumber(),
+                                                                                       convert(version.confidence()));
+                }
+                // TODO jonmv: Remove try/catch when this is supposed to work.
+                catch (Exception e) {
+                    log.log(Level.INFO, "Failed reporting system version status to monitor: " + Exceptions.toMessageString(e));
+                }
+            });
         } catch (Exception e) {
             log.log(Level.WARNING, "Failed to compute version status: " + Exceptions.toMessageString(e) +
                                    ". Retrying in " + interval());
         }
     }
 
+    private static SystemMonitor.Confidence convert(VespaVersion.Confidence confidence) {
```

Comment:
Add a test that verifies that all `VespaVersion.Confidence` can be converted?

Reply:
Fair enough.